clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -532,15 +532,21 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
                     try {
                         channel.prepare();
                     } catch (AuthenticationException e) {
-                        if (channel.successfulAuthentications() == 0)
+                        boolean isReAuthentication = channel.successfulAuthentications() > 0;
```

Comment:
Nit: capitalization isn't quite right, see `reauthenticationLatencyMs` for example.